docs: add ip address guidance for partners (#176)
Gusto/embedded-react-sdk: docs/03/getting-started-1.md

export default App
```

The `baseUrl` property is configured with the address of your backend proxy which is detailed further in the following section.

## Configuring a backend proxy

When building with the React SDK, a backend proxy is required. React SDK components do not make calls to the Gusto Embedded API directly. Instead, the `baseUrl` configuration defines the URL of your proxy server. This proxy layer gives you complete control over requests sent to Gusto, which is essential for:

1. Authentication
2. Providing the user IP address for form signing operations

The React SDK is designed to mirror the [Gusto Embedded API Reference](https://docs.gusto.com/embedded-payroll/reference/whats-new-in-v2024-04-01) with a 1:1 mapping of endpoints. The SDK maintains consistent naming conventions, parameters, and response structures with the Gusto API.

> Your proxy server simply needs to forward incoming SDK requests to the corresponding Gusto API endpoints. No special endpoint mapping or request transformation is required, as the SDK requests already match the Gusto API format. The proxy's main task is adding the necessary authentication headers before forwarding the request.

Reviewer: ```suggestion
Your proxy server simply needs to forward any incoming SDK requests to the corresponding Embedded API endpoints. The proxy's main task is adding the necessary authentication headers before forwarding the request onwards. Since the SDK requests are already in the Embedded API format, no extra endpoint mapping or request transformation is required. 
```

### Using the proxy for authentication

> The proxy layer allows for authentication. The recommended approach is to use a backend service that acquires OAuth2 tokens from the Gusto Embedded API for authenticated users and proxies API calls using those tokens. Learn more about configuring this and setting up authentication in the `Authentication` section.

Reviewer: General question: what's the recommended approach for accessing different company access tokens or refresh tokens via the proxy? 

It's super clear on `/company/{companyID}` endpoints; but how to handle for non company based routes like `/employees/{employee_id}/home_addresses`?  Say I have 10 partner managed companies.

Author: Good question! This is meant to be an extremely lightweight explanation that authentication will happen in the proxy layer and not the SDK. The reference there for learning more about configuring authentication is located here https://docs.gusto.com/embedded-payroll/docs/authentication-1 Lmk if that doc does a better job of clarifying, if not we can probably create a separate ticket to update that page with more details

Reviewer: More details to come on this, resolving for now.

### Using the proxy to provide the user IP address

> Some UI workflows require users to sign forms, which need the user's IP address for security purposes. Since the SDK runs in the browser, it cannot reliably access the IP address.

Reviewer 2: I think it might be worth calling out more that the client could spoof the IP instead of the lack of the browsers ability to get the IP. Because I think there are many ways to get the IP but none of them provide the IP in a way which can't be spoofed.

Author: Good callout! updated the wording, lmk if that accounts for the above

Reviewer 2: Looks good!

> Some UI workflows require users to sign forms, which need the user's IP address for security purposes. To prevent vulnerabilities such as IP address spoofing, this information must be provided by your proxy server rather than collected client-side.

Reviewer: So the partner's server IP address is fine?

Author: They would be retrieving the user IP address from the request and then setting it via the `x-gusto-client-ip` header. Lmk if there's better wording here to describe that! attempted to explain further below

Reviewer: Sounds good, it is on the partner to retrieve the clients IP address possibly when the SDK is launched or access.
